### ISSUE TITLE
Submit TheGroceryForm via vuex approach

### ISSUE THREAD

**Issue by brianzelip** [brianzelip/groceries-vue#11]
Be more consistent with handling user data.

**tl;dr** This is a vuex app!

Most of the work done in the `GroceryForm/` components is consistent when it comes to updating state of the user input form data. This means that, *instead of passing data down to child components, and from children up to parent components via custom events, the components that contain the actual UI that that user interacts with to input data is the component that triggers the state change*.

The one action where this is not consistent, in the GroceryForm side of things, is when the form submit button is clicked. Instead of `<TheGroceryFormControls>` performing the submit act, it emits [an event for `<TheGroceryForm>` to submit](https://github.com/brianzelip/groceries-vue/blob/4905e71144562ae90bfdea677697f88064e19267/src/components/GroceryForm/TheGroceryForm.vue#L36).

Although this form submission does not change state, I have a hunch that I should follow suit with the way in which state is updated, and perform the submit form method from the actual UI component that is responsible for handling that particular user interaction.

Maybe this isn't necessary though. But the thing is, after working on my concats app for a while, I got used to passing data around via props and custom events. So my recent work in `ItemForm/` components has taken on that type of state change. I should work to be more consistent throughout the app!
